tests/rptest/tests/pandaproxy_test.py

```diff
@@ -2235,3 +2238,113 @@ def do_coordinator_change(utils: GroupCoordinatorTransferUtils):
         self.logger.debug(
             "Test offset commit: after second coordinator change")
         do_consumer_offset_commit(new_offset=1)
+
+
+class PandaProxyCompressedBatchesTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(partition_count=1),
+    ]
+
+    def __init__(self, context):
+
+        super(PandaProxyCompressedBatchesTest,
+              self).__init__(context,
+                             extra_rp_conf={
+                                 "enable_leader_balancer": False,
+                                 "group_topic_partitions": 1
+                             })
+
+    def random_printable_str(self, size: int = 10):
+        return "".join(
+            random.choice(string.ascii_lowercase) for _ in range(size))
+
+    def _produce_with_compression(self, num_bytes: int, compression_type: str,
+                                  partition: int):
+        rpk = RpkTool(self.redpanda)
+        msg = self.random_printable_str(num_bytes)
+        offset = rpk.produce(self.topic,
+                             key='k',
+                             msg=msg,
+                             compression_type=compression_type)
+        self.logger.debug(f'RPK produced to offset {offset}')
+        return msg, offset
```

Comment:
Since the test requires a compressed batch, I think the `rpk.produce` call should produce a known canned message that always compresses (something like 1024 'a's).  This is so on the off chance that the message generated by `random_printable_str` doesn't compress, this doesn't cause a unnecessary test failure.